packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -301,34 +304,16 @@ class SystemChrome {
   /// }
   /// ```
   static void setSystemUIOverlayStyle(SystemUiOverlayStyle style) {
-    assert(style != null);
-
-    if (_pendingStyle != null) {
-      // The microtask has already been queued; just update the pending value.
-      _pendingStyle = style;
-      return;
-    }
-
-    if (style == _latestStyle) {
-      // Trivial success: no microtask has been queued and the given style is
-      // already in effect, so no need to queue a microtask.
+    if (style == _latestStyle || style == null) {
       return;
     }
+    _latestStyle = style;
 
-    _pendingStyle = style;
-    scheduleMicrotask(() {
-      assert(_pendingStyle != null);
-      if (_pendingStyle != _latestStyle) {
-        SystemChannels.platform.invokeMethod(
-          'SystemChrome.setSystemUIOverlayStyle',
-          _pendingStyle._toMap(),
-        );
-        _latestStyle = _pendingStyle;
-      }
-      _pendingStyle = null;
-    });
```

Comment:
we might want to keep the old behaviour around in case anyone else wants to use this API, and doesn't have a way to guarantee they only call it once per frame.